oat-sa/extension-tao-itemqti-pci
# fix/AUT-2552/keep-mapping-default-to-zero

**Related to:** [AUT-2552](https://oat-sa.atlassian.net/browse/AUT-2552)

**Description:**
Due to a misunderstanding, the Mapping Default value is 1 now, which makes that incorrect response take this default value of 1, so it has a positive score.
Tasks:
- Keep the Mapping Default value to 0 for the incorrect responses. 
- Set a default value of 1 to the correct responses created on the Authoring Response mode.
- Keep the binding between the Mapping Default value and the correct score value when the value differs from 0. Otherwise, it will be 1.

**Changes:**

- Create a new variable with a default value of 1 that will be used as the default value for the score of correct responses. Keeping separated from the Mapping Default value.
- Update this variable if the Mapping Default value differs from 0.

**How to check:**

- Add MathEntry PCI to the item. Add some correct responses. Save it and check on Preview that non-correct responses have a Score of 0.
- Change the Mapping Default to a value different from 0 and check if the non-correct response has the Score of the Mapping Default value.
- Check on Authoring responses mode that changing the Mappgin Default value affects new alternative responses. Change back to 0 and check that the new alternative responses have a value of 1.
